File: AccountBook/Java-AccountBook/src/AccountBook/Outgo.java
package AccountBook;

public class Outgo extends Account
{
    //디폴트생성자
    public Outgo()
    {
        super();
    }
    //매개변수를 가지는 생성자
    public Outgo(Date date, String briefs, int amount, int balance, String remarks)
    {
        super(date, briefs, amount, balance, remarks);
    }
    //clone 오버라이딩
    @Override
    public Account clone()
    {
        Date date = new Date(this.date.getYear(), this.date.getMonth().getValue(),
                this.date.getDay());
        return new Outgo(date, this.briefs, this.amount, this.balance, this.remarks);
    }

    //잔액(balance) 계산하기 메소드 오버라이딩
    @Override
    public int calculateBalance(Account account)
    {
        return account.balance - this.amount;
    }
    @Override
    public int calculateBalance(int amount)
    {
        return this.balance + (this.amount + amount);
    }

    //setAmount 오버라이딩
    @Override
    public void setAmount(int amount)
    {
        this.amount = amount * (-1);
    }
}

File: AccountBook/CPlusPlus-AccountBook/CPlusPlus-AccountBookFinalVersion/Outgo.cpp
#include "Outgo.h"

//디폴트생성자
Outgo::Outgo()
{

}

//매개변수5개생성자
Outgo::Outgo(Date date, string briefs, Currency amount, Currency balance, string remarks)
	:Account(date, briefs, amount, balance, remarks)
{

}

//복사생성자
Outgo::Outgo(const Outgo& source)
	:Account(source)
{

}

//치환연산자
Outgo& Outgo::operator=(const Outgo& source)
{
	Account::operator=(source);

	return *this;
}

//소멸자
Outgo::~Outgo()
{

}